BlockSettleSigner/SignerAdapterListener.cpp

```diff
@@ -337,8 +337,34 @@ bool SignerAdapterListener::onSignOfflineTxRequest(const std::string &data, bs::
       if (txSignReq.walletIds.size() == 1) {
          const auto &wallet = walletsMgr_->getWalletById(txSignReq.walletIds.front());
          bs::core::WalletPasswordScoped lock(hdWallet, SecureBinaryData::fromString(request.password()));
-         const auto tx = wallet->signTXRequest(txSignReq);
-         evt.set_signedtx(tx.toBinStr());
+
+         bool isLedgerHw = false;
+         if (hdWallet->isHardwareWallet()) {
+            bs::wallet::HardwareEncKey hwEncKey(hdWallet->encryptionKeys()[0]);
+            isLedgerHw = hwEncKey.deviceType() == bs::wallet::HardwareEncKey::WalletType::Ledger;
+         }
+
+         BinaryData signedTx;
+         if (isLedgerHw) {
+            // For ledger hw data is not prepared straight away
+            Blocksettle::Communication::headless::InputSigs sigs;
+            if (!sigs.ParseFromString(SecureBinaryData::fromString(request.password()).toBinStr())) {
```

Comment:
`request.password()` is used for 2 different tasks simultaneously: for parsing Ledger sigs and for locking a wallet with `WalletPasswordScoped`. I don't think this is intended. Can you add more OOD and encapsulation here?
I believe `SignerAdapterListner` shouldn't know anything about Ledger specifics - everything should happen under the `wallet->signTXRequest()` hood. Maybe wallet should be subclassed to serve Ledger-signed transactions. Or `WalletPasswordScoped` should be subclassed to handle Ledger sigs parsing instead of decrypting private key.

Reply:
After discussion with Sergey, we decided:
1. Do not change location of sig sign (e.g. use the same password container for both sign sig and actually password), because of amount of work
2. Sign will be regulated by hdWallet, since it knows structure of child leaf and contain all needed information to distinguish which type of sign should be used.
Thanks

Reply:
Fixed, also I have added changes to common to fix this flow - BlockSettle/common#1928, thanks